gcc5 gives array-bounds warning on xen/common/symbols.c
     also maybe-uninitialized warning on xen/xsm/flask/ss/policydb.c

--- xen-4.5.0/xen/common/Makefile.orig	2015-01-12 16:53:24.000000000 +0000
+++ xen-4.5.0/xen/common/Makefile	2015-02-14 15:40:29.722759007 +0000
@@ -72,3 +72,5 @@
 
 subdir-y += libelf
 subdir-$(HAS_DEVICE_TREE) += libfdt
+
+CFLAGS += -Wno-error=array-bounds
--- xen-4.5.0/xen/xsm/flask/Makefile.orig	2015-01-12 16:53:24.000000000 +0000
+++ xen-4.5.0/xen/xsm/flask/Makefile	2015-02-14 16:49:54.376183206 +0000
@@ -5,6 +5,7 @@
 subdir-y += ss
 
 CFLAGS += -I./include
+CFLAGS += -Wno-error=maybe-uninitialized
 
 AWK = awk
 
--- xen-4.6.0/tools/blktap2/drivers/Makefile.orig	2015-08-08 19:56:57.000000000 +0100
+++ xen-4.6.0/tools/blktap2/drivers/Makefile	2015-08-08 20:59:41.112636845 +0100
@@ -18,6 +18,7 @@
 CFLAGS    += -DUSE_NFS_LOCKS
 # drivers/block-log.c incorrectly uses libxc internals
 CFLAGS    += -I$(XEN_ROOT)/tools/libxc
+CFLAGS += -Wno-error=array-bounds
 
 ifeq ($(CONFIG_X86_64),y)
 CFLAGS            += -fPIC
